pkg/hive/hive.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hive
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ethersphere/bee/pkg/hive/pb"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+)
+
+const (
+	protocolName           = "hive"
+	initStreamName         = "hive_init"
+	initStreamVersion      = "1.0.0"
```

Comment:
This may be a nice simplification, lets try to see if it is easy to change the stream template `/swarm/{protocolName}/{protocolVersion}/{streamName}`, instead `/swarm/{protocolName}/{streamName}/{streamVersion}`.